frontend/simple/js/event-log.js

```diff
@@ -0,0 +1,134 @@
+import localforage from 'localforage'
+import { toHash, makeEntry, makeGroup } from '../../../shared/functions'
+import request from 'superagent'
+
+// =======================
+// Helpers
+// =======================
+function getDB (hash) {
+  let db = localforage.createInstance({
+    name: 'Group Income',
+    storeName: hash
+  })
+  return db
+}
+function postEvent (groupId, hash, entry) {
+  return new Promise(function (resolve, reject) {
+    request.post(`${process.env.API_URL}/event/${groupId}`)
+      .send({hash, entry})
+      .end(function (err, res) {
+        if (err) {
+          return reject(err)
+        }
+        return resolve()
+      })
+  })
+}
+function postGroup (hash, entry) {
+  return new Promise(function (resolve, reject) {
+    request.post(`${process.env.API_URL}/group`)
+      .send({hash, entry})
+      .end(function (err, res) {
+        if (err) {
+          return reject(err)
+        }
+        return resolve(res.hash)
+      })
+  })
+}
+// =======================
+// Exports
+// =======================
+// get retrieves from the log
+// opts object allows for parent hash to be returned instead of data
+export async function get (log, opts) {
```

Comment:
This needs a more descriptive name. I also recommend changing the function signature to something like:

```js
export async function getItemFromLog(logId, key, opts) {
...
}
```

You will also be able to remove (or significantly shorten) the `if`-blocks that check opts. Use ES6 destructuring on the `opts` object also (if it's defined).